### PR TITLE
ci: automatic rebase

### PR DESCRIPTION
**Pull request by josokinas**
## Purpose

Adding ability to rebase automatically.

## Approach

Based on [GitHub action to automatically rebase PRs](https://github.com/cirrus-actions/rebase).

## Testing

Will need to test on future PR, unfortunately.

## Risks

N/A

### CODE REVIEW

**Comment by rabelloo** [comment]
Please help me understand why this is needed.

Can't we simply use GitHub's Rebase button?

image

**Comment by josokinas** [comment]
> Can't we simply use GitHub's Rebase button?

GitHub does not allow to rebase on UI without merging, how would you do this?

**Comment by rabelloo** [comment]
But why would we want to rebase if not for merging? 🤔

**Comment by josokinas** [comment]
I feel like it's handy especially for older pull requests, open source contributions? Just to sync and see how's pipeline reacting before any further actions.

**Comment by rabelloo** [comment]
Cool. I guess we can give it a try.

File: .github/workflows/rebase.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Rebase
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  rebase:
+    name: Automatic rebase
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/rebase')
```
Comment (rabelloo) [inline .github/workflows/rebase.yml:10]: Can we change this to a full equality match lest we accidentally rebase it with other comments?
e.g. `just do a /rebase`.

```suggestion
    if: github.event.issue.pull_request != '' && github.event.comment.body == '/rebase'
```

**Review by rabelloo — CHANGES_REQUESTED** [review]

**Review by rabelloo — APPROVED** [review]

**Pull request merged**